compiler/rustc_borrowck/src/lib.rs

```diff
@@ -522,15 +518,15 @@ impl<'tcx> Deref for BorrowckInferCtxt<'tcx> {
     }
 }
 
-struct MirBorrowckCtxt<'a, 'mir, 'infcx, 'tcx> {
+struct MirBorrowckCtxt<'mir, 'infcx, 'tcx> {
     infcx: &'infcx BorrowckInferCtxt<'tcx>,
     param_env: ParamEnv<'tcx>,
     body: &'mir Body<'tcx>,
-    move_data: &'a MoveData<'tcx>,
+    move_data: &'mir MoveData<'tcx>,
```

Comment:
Same here, `'mir` is wrong, if you want to merge the lifetimes, move to `'a` which is what we had before my refactorings

Reply:
I wrote about this in the first commit message:

> The commit uses the name `'mir` everywhere for the lifetimes used with
`&Body` and `&MoveData`, replacing some other names: `'a`, `'flow`, and
`'cx`. I think this is a reasonable name because `'mir` is used for many
`mir::Body` references (and things within `mir::Body`). Even when a
lifetime is shared between a `&Body` and a reference to something else,
the `'mir` lifetime name is so well-established that it's reasonable to
reuse it, rather than something non-specific like `'a`. (Naming
lifetimes is hard!)

I considered using `'a`, but one downside is if you have `Foo<'a, 'tcx>` and then you put a reference to that inside something else you get `&'b Foo<'a, 'tcx>`? Or maybe `&'a Foo<'b, 'tcx>`? And then if you put a reference to that inside something else again you get maybe `&'a Bar<'b, 'c, 'tcx>`?